src/components/views/messages/TextualBody.js

```diff
@@ -80,6 +93,20 @@ module.exports = React.createClass({
                     }
                 }, 10);
             }
+            const buttons = ReactDOM.findDOMNode(this).getElementsByClassName("mx_EventTile_copyButton");
+            if (buttons.length > 0) {
+                // Do this asynchronously: parsing code takes time and we don't
+                // need to block the DOM update on it.
+                setTimeout(() => {
+                    if (this._unmounted) return;
+                    for (let i = 0; i < buttons.length; i++) {
+                        buttons[i].onclick = (e) => {
+                            const copyCode = buttons[i].parentNode.getElementsByTagName("code")[0];
+                            this.copyToClipboard(copyCode.textContent);
+                        };
+                    }
+                }, 10);
+            }
```

Comment:
It took me a while to realise that of course you can't add the onclick handlers here because you're just using it to parse, manipulate and re-generate the HTML markup. I'm going to add a comment to make it a bit easier to understand what's going on here, but otherwise this looks good.